src/com/google/javascript/jscomp/BranchCoverageInstrumentationCallback.java

```diff
@@ -89,11 +94,16 @@ public void visit(NodeTraversal traversal, Node node, Node parent) {
       for (DiGraph.DiGraphEdge<Node, ControlFlowGraph.Branch> outEdge : cfg.getOutEdges(node)) {
         if (outEdge.getValue() == ControlFlowGraph.Branch.ON_FALSE) {
           Node destination = outEdge.getDestination().getValue();
-          if (destination.isBlock()) {
+          if (destination != null && destination.isBlock()) {
             blocks.add(destination);
           } else {
             Node exitBlock = IR.block();
-            destination.getParent().addChildBefore(exitBlock, destination);
+            if (destination != null && destination.getParent().isBlock()) {
+              destination.getParent().addChildBefore(exitBlock, destination);
+            } else {
+              outEdge.getSource().getValue().getParent()
```

Comment:
Personally when I'm reading compiler code it's useful to describe each case in comments as what exactly it handles. Otherwise it's hard to visualize AST tree here by just looking at checks in `if` statements.

Reply:
Updated.